src/EnergyPlus/HeatBalanceIntRadExchange.cc

```diff
@@ -1463,22 +1461,24 @@ namespace HeatBalanceIntRadExchange {
         // allow that the sum of all view factors from the original surface to all other
         // surfaces will equal unity.  F(I,J)=0 if I=J or if the surfaces face the same
         // direction.
-        //  Modified to use Class of surface to permit INTMASS to be seen by all surfaces,
-        //  FLOOR to be seen by all except other floors, and ROOF and CEILING by all.
-        // The second IF statement is intended to avoid a divide by zero if
-        // there are no other surfaces in the zone that can be seen.
         F = 0.0;
         for (i = 1; i <= N; ++i) {
             for (j = 1; j <= N; ++j) {
-
-                //  Skip same surface
-
-                if (i == j) continue;
-                //  Include INTMASS, FLOOR(for others), CEILING/ROOF  and different facing surfaces.
+                //  No surface sees itself and no floor sees another floor
+                if (i == j || (state.dataSurface->Surface(SPtr(j)).Class == SurfaceClass::Floor &&
+                               state.dataSurface->Surface(SPtr(i)).Class == SurfaceClass::Floor))
+                    continue;
+                // All surface types see internal mass
+                // All surface types see floors
+                // Floors always see ceilings/roofs
+                // All other surfaces whose tilt or facing angle differences are greater than 10 degrees see each other
                 if ((state.dataSurface->Surface(SPtr(j)).Class == SurfaceClass::IntMass) ||
+                    (state.dataSurface->Surface(SPtr(i)).Class == SurfaceClass::IntMass) ||
```

Comment:
@mjwitte FYI, this is what was missing previously. All surfaces would see internal mass, but internal mass wouldn't see other surfaces.